.github/workflows/test.yml

```diff
@@ -32,24 +33,31 @@ jobs:
       - name: Inject slug/short variables
         uses: rlespinasse/github-slug-action@v4
 
-      - name: Set up gcloud
+      # Setup gcloud CLI
+      - name: Authenticate to Google Cloud
+        id: auth
+        uses: google-github-actions/auth@v0.5.0
+        with:
+          credentials_json: ${{ secrets.GOOGLE_CREDENTIALS }}
+
+      - name: Set up gcloud SDK environment
         uses: google-github-actions/setup-gcloud@v0.4.0
         with:
           project_id: ${{ env.PROJECT_ID }}
-          service_account_key: ${{ secrets.GCLOUD_AUTH }}
 
       # Creates Compute Engine virtual machine instance w/ disks
       - name: Create instance
         run: |
           gcloud compute instances create-with-container "zebrad-tests-${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}-${{ env.GITHUB_SHA_SHORT }}" \
           --boot-disk-size 100GB \
           --boot-disk-type pd-ssd \
+          --container-stdin \
+          --container-tty \
           --container-image rust:buster \
           --container-mount-disk mount-path='/mainnet',name="zebrad-cache-${{ env.GITHUB_SHA_SHORT }}-mainnet-canopy" \
           --container-restart-policy never \
           --create-disk name="zebrad-cache-${{ env.GITHUB_SHA_SHORT }}-mainnet-canopy",image=zebrad-cache-1558f3378-mainnet-canopy \
```

Comment:
The disk name is changed in the regenerate cache workflow, but not in the cached test workflow.

Can we regenerate state and update the cached disk name before merging this PR?
That seems like a good test that everything works.

Reply:
I'll remove all the changes outside the scope of `cd.yml` and keep those in this PR: 
 https://github.com/ZcashFoundation/zebra/pull/3459
 
 This will help on focusing solely on continuous deployment, instead of testing. As I've advanced that part even more there.

Reply:
`regenerate-stateful-test-disks.yml` is still changing the disk names in this PR, can we move it to PR #3459 as well?

Reply:
Fixed